driver/lock.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"github.com/Sirupsen/logrus"
+	ibclient "github.com/infobloxopen/infoblox-go-client"
+	"time"
+)
+
+type GridLocker struct {
+	timeout time.Time
+	objMgr  *ibclient.ObjectManager
+	name    string
+	ref     string
+}
+
+func (l *GridLocker) Lock() bool {
+	// create EA definition with name as in lockName variable
+	// If EA is created successfully, then lock is acquired
+	// If response has an error, then lock is held by some other resource
+
+	logrus.Debugf("Creating lock %s\n", l.name)
+
+	eaDef := ibclient.EADefinition{Name: l.name, Type: "STRING", Flags: "C",
+		Comment:      "Docker Resource Lock",
+		DefaultValue: string(time.Now().Unix())}
+
+	// TODO: implement timeout so that lock is never held infinitely for a resource
+	// This can happen when a system which acquired lock crashes without releasing the lock
+
+	eaDefRef, err := l.objMgr.CreateEADefinition(eaDef)
```

Comment:
A couple of things:

1) Create EA Definition requires "Superuser" permission. We should try to accomplish the same using EA instead.

2) You might want to have a timestamp as part of the persisted GridLock object, so the code can recover from situation if a lock has not been released due to an exception. i.e. if the timestamp of a lock is beyond a certain age, the code can assume it's a stale lock.